dist2src/core.py

```diff
@@ -412,7 +447,44 @@ def rebase_patches(self, from_branch, to_branch):
             self.source_git.checkout(from_branch)
             self.source_git.repo.git.reset("--hard", "HEAD^")
 
-        self.source_git.repo.git.rebase("--root", "--onto", to_branch)
         self.source_git.checkout(to_branch)
-        self.source_git.repo.git.merge("--ff-only", "-q", from_branch)
-        self.source_git.repo.git.branch("-d", from_branch)
+        commits_to_cherry_pick = [
+            c.hexsha[:8]  # shorter format for better readability in case of an error
+            for c in self.source_git.repo.iter_commits(from_branch)
+        ][-2::-1]
+        if commits_to_cherry_pick:
+            self.source_git.repo.git.cherry_pick(
+                *commits_to_cherry_pick,
+                keep_redundant_commits=True,
+                allow_empty=True,
+                strategy_option="theirs",
+            )
+        self.source_git.repo.git.branch("-D", from_branch)
+
+    def update_source_git(self, origin_branch: str, dest_branch: str):
+        """
+        Update the existing source-git.
+
+        1. Revert the patches.
+        2. Convert the dist-git to source-git
+        3. Fast-forward the branch
+
+        :param origin_branch: branch used as a dist-git source
+        :param dest_branch: source-git branch we need to update
+        """
+        self.dist_git.clean()
+        self.dist_git.checkout(branch=origin_branch)
+
+        dg_tags_for_head = self.dist_git.get_tags_for_head()
+        new_dest_branch = (
+            dg_tags_for_head[0]
+            if dg_tags_for_head
+            else f"{dest_branch}-{self.dist_git.repo.head.commit.hexsha:.8}"
+        )
+        self.source_git.checkout(dest_branch)
+        self.source_git.checkout(branch=new_dest_branch, create_branch=True)
+        self.source_git.revert_to_ref(START_TAG, message="Prepare for a new update")
```

Comment:
+1 nice, you saw my comment in the etherpad :+1: 

we could append `'\n\nReverting patches so we can apply the latest update\nand changes can be seen in the spec file'`

but I guess that we can document the reason for this particular commit.

Reply:
Fixed.